core/include/drivers/imx_i2c.h

```diff
@@ -9,6 +9,38 @@
 #include <stdint.h>
 #include <tee_api_types.h>
 
+#if defined(CFG_MX8MM)
+#include <registers/imx8m.h>
+/* clock configuration */
+#define I2C_CLK_CGR(__x)	CCM_CCRG_I2C##__x
+/* iomux configuration */
+#define IOMUXC_I2C1_SCL_CFG_OFF	0x47C
```

Comment:
I think you misread my comment https://github.com/OP-TEE/optee_os/pull/4152#discussion_r511817381 
I only asked if you could move ```I2C_``` definitions, not ```IOMUXC```

 ```IOMUXC``` can definitly go to ```imx8m.h``` header file

Reply:
It is a bit confusing because you are asking for offsets that are defined as IOMUXC_I2C1_SCL_CFG_OFF in imxX.h but offsets defined using macros (ie for I2C2 and I2C3) in a different file which is counter intuitive.

And we are also deviating from what was asked when support for imx8m was merged.

if it was down to me I'd still write:
1) offsets in imx8m.h 
2) hardcoded target values for clock and iomux (since we dont have those drivers)
 - either in the corresponding iomux.h and crm.h 
 - or even better just in the imx_i2c.c driver itself

anyway, I'll try again

Reply:
@clementfaure I am going to close this PR and open a new one from which I hope to get some clarity on the review comments (some of the things you are asking don't quite make much sense to me).

**1) moving I2C_CFG_ to imx_i2c.h** 

why? it does not belong in imx_i2c.h...perhaps in imx_i2c.c since some I2C_ are utility macros; however not all of them are: others are hardcoded clock and iomux calculated values...so where do you expect these later values to land? and why cant we do now as it was done for imx8mm.?

2)  importing an imx6-iomux.h 
why? the i2c driver will not use any values from that file so why do we need it done in this PR.

3) dropping support for boot-loaders not setting the peripheral clock to 24MHz
why? can you provide a valid reason for that.

so please bear me, let me send the PR again, and then we can discuss and hopefully align.

Reply:
@ldts 
Alright, no problem, I will have a look at the new PR.
I suggest we find a solution in the comment section that suit both of us first, and then push changes. By doing so, I believe we would avoid confusion and new pull-requests. What do you think?

**1/ Moving ```I2C_CFG_``` to ```imx_i2c.h```**
I believe ```I2C_CFG_``` macros should belong to ```imx_i2c.h``` as the name of the macro suggest, same thing for ```IOMUXC_``` and ```imxX-iomux.h```. On the ```imx8m``` we don't have ```imx8m-iomux.h```, that's why I was okay with mixing ```I2C_CFG_``` and ```IOMUXC_```. Maybe I'm being a bit picky here, I don't know. I won't request any change for this on the new pull request.

**2/  Importing an imx6-iomux.h**
I asked you to **not** import anything from Code Aurora https://github.com/OP-TEE/optee_os/pull/4152#discussion_r511820100 but instead move existing ```IOMUXC_``` definitions from ```imx6.h``` to ```imx6-iomux.h```.
You noticed that many of definitions to be moved are unused : alright, let's remove them! https://github.com/OP-TEE/optee_os/pull/4152#discussion_r512504625 So we would end up with only i2c related definitions in  ```imx6-iomux.h```.
I will request the change on the new PR.

**3) dropping support for boot-loaders not setting the peripheral clock to 24MHz**
I provided the reason here : https://github.com/OP-TEE/optee_os/pull/4152#discussion_r512527245
Btw, could you answer my questions about this clock rate ? https://github.com/OP-TEE/optee_os/pull/4152#discussion_r512503493

See you on the other PR!

Reply:
I am getting so sick of github :) ...it is impossible to have a coherent conversation.

So to answer on the 24MHz, it is not something that I need for SE050 (we merged the patch I sent to u-boot to correct the clock issue so we are already ok in our baselines using the 24MHz clock). but I was thinking of other users that are booting with these old versions of u-boot....should not we be concerned about them?


btw u-boot are merging this patch so from the next release, the clock upstream will be 24MHz on imx6ull